common/ipfs/api.go

```diff
@@ -21,15 +22,38 @@ const defaultIpfsLimiterMaxQueueDepth = 100
 const defaultIpfsPublishPubsubTimeoutS = 30 * time.Second
 
 type IpfsApi struct {
-	core            iface.CoreAPI
-	logger          models.Logger
-	multiAddressStr string
-	metricService   models.MetricService
-	limiter         *ratelimiter.RateLimiter[any, any]
+	core          iface.CoreAPI
+	logger        models.Logger
+	addrStr       string
+	metricService models.MetricService
+	limiter       *ratelimiter.RateLimiter[any, any]
 }
 
-func NewIpfsApiWithCore(logger models.Logger, multiAddressStr string, coreApi iface.CoreAPI, metricService models.MetricService) *IpfsApi {
-	ipfs := IpfsApi{core: coreApi, logger: logger, multiAddressStr: multiAddressStr, metricService: metricService}
+func createCoreApi(addrStr string) (*rpc.HttpApi, error) {
+	addr, err := ma.NewMultiaddr(addrStr)
```

Comment:
So this approach will use the IP address instead of the multiaddr? If so, I actually like this better? Keeps things simple. We could definitely retrieve the peer ID from Keramik and feed it in, but IP address is easier and more "functional", IMO.

Reply:
yes it does haha i don't know why i didn't see this option before. The break has given me fresh eyes =)